### ISSUE TITLE
Cannot install extensions referenced by a github link to a tarball, only zipball links work

### ISSUE THREAD

**Issue by Lucas-C** [splitbrain/dokuwiki#1296]
Hi,

I found out about this while trying to install "dokufreaks/plugin-include" from the Extension Manager.
I got an error stating "Unable to identify extension directory, you need to download and install manually".

Then I found out this issue only arise with an extension where the "install" button point to a tarball, not for an extension where the link points to a zipball.

I suspect that, at this line of code:
https://github.com/splitbrain/dokuwiki/blob/master/lib/plugins/extension/helper/extension.php#L870
an silent error occur during the archive decompression.

This recent commit is probably the culprit :)
https://github.com/splitbrain/dokuwiki/commit/605f8e8d0e501057749c50581087ce05089c1af3#diff-441b0afb846829ac54893372ccbaa72c

The same error has been observed in this issue, where the link point to a tarball:
https://github.com/michitux/dokuwiki-plugin-outliner/issues/17

**Comment by Lucas-C** [comment]
With some traces I found out the extension is guessed to be 'gz'.
But are tar.gz archives supported ?

**Comment by Klap-in** [comment]
The archive library used by Dokuwiki is today: https://github.com/splitbrain/php-archive
It is a rewrite of the previous one, sofar I know it should handle .tar.gz fine.
So maybe somewhere an bug is introduced?

**Comment by splitbrain** [comment]
See https://github.com/splitbrain/php-archive/issues/10

From what I see so far, github's tar files are wrong. But I'm still investigating.

**Comment by splitbrain** [comment]
Okay it get's weirder. I can download that file and extract it with the Tar class without any problems. The downloaded file has the same MD5 as the one DokuWiki downloads, but within DokuWiki the header checksum fails.

**Comment by Lucas-C** [comment]
Have you checked that the extracted directory is not empty ?
Because I oberserved that.

Also, I was able to `tar -xzvf` the github tarball without any issue

**Comment by splitbrain** [comment]
Found the problem. Since the file is named ```master``` the library fails to guess the compression type correctly. Fix is in the work

**Comment by gamma** [comment]
Are there any updates on this issue? I think this is a serious one a and should justify a hot fix.

**Comment by splitbrain** [comment]
Yeah. Maybe a hotfix would be in order. Can you suggest it on the mailing list?

**Comment by Lucas-C** [comment]
Thanks for the fix !

**Comment by LarsGit223** [comment]
Is this really fixed?

I'm on Release 2015-08-10a "Detritus" and noticed today that I can not install the Creole plugin. I changed the link on the page to **zipball** and then it worked.

Also I have to mention the extension manager did not give any error message after installing the plugin. It said installation was successful. But I noticed the Creole plugin didn't work. I wanted to check the plugin configuration and noticed the Creole plugin was not listed on the config page. Then I did go back to the extension manager again and noticed the plugin was installed but disabled. I clicked on **enable** but the page just re-loaded and the **enable** button was still shown. That means I could not enable the plugin.

As mentioned above, everything worked fine as soon as I changed the download link to zipball.
